### ISSUE TITLE
Nested transactions with savepoints don't execute callbacks when expected

### ISSUE THREAD

**Issue by pabloh** [chanks/sequel-savepoint-hooks#2]
```ruby
  describe "when a transaction is opened with hooks: :savepoint" do
    it "should open savepoints inside all the nested transaction with callbacks" do
      DB.transaction hooks: :savepoint do
        DB.after_commit   { @hooks << :commit1   }
        DB.after_rollback { @hooks << :rollback1 }

        DB.transaction savepoint: true do
          DB.after_commit   { @hooks << :commit2   }
          DB.after_rollback { @hooks << :rollback2 }

          DB.transaction savepoint: true do
            DB.after_commit   { @hooks << :commit3   }
            DB.after_rollback { @hooks << :rollback3 }
          end
        end

        # It fails: only contains [:commit2] instead
        assert_equal [:commit3, :commit2], @hooks 
      end

      assert_equal [:commit3, :commit2, :commit1], @hooks
    end
  end
```

**Comment by chanks** [comment]
Interesting, maybe due to a change in the Sequel gem.

**Comment by pabloh** [comment]
Does it work on older versions?, I have only tried it on the 5.x.x series, I'll push a PR with the tests anyway.

**Comment by chanks** [comment]
Oh, is this new behavior you're requesting? I assumed you were pointing out that a spec is now failing.

I haven't looked at this gem in forever and I don't think I currently have any projects that use it - I'd have to do some digging to figure out how/if to implement this.

**Comment by pabloh** [comment]
I run the tests on `4.35.0` and had the same error.

**Comment by pabloh** [comment]
> Oh, is this new behavior you're requesting? I assumed you were pointing out that a spec is now failing.
> 
> I haven't looked at this gem in forever and I don't think I currently have any projects that use it - I'd have to do some digging to figure out how/if to implement this.

Ok, well let me know if this is behavior you think the gem should support, maybe I can help with a patch if we agree on the specs.


Is a shame you are not using it anymore, did you find a better solution or just not using Sequel nowadays?

**Comment by pabloh** [comment]
Is a shame you are not using it anymore, do you found a better solution or just not using Sequel anymore?

**Comment by chanks** [comment]
Nah, I still use Sequel, I just moved away from using after_commit. It was mainly useful for sending work to a non-Postgres job queue, but as I've transitioned things to [Que](https://github.com/chanks/que) it hasn't been necessary.

Anyway, I think that it makes sense for the gem to follow Sequel's semantics. So when you nest a savepoint inside another one, does Sequel use different names for each one, support rolling back the inner one but not the outer one, and so on? If so then yes, I think the behavior you're looking for makes sense.

**Comment by pabloh** [comment]
Nice, I'll put together a PR.
